yarn-project/cli-wallet/src/cmds/index.ts

```diff
@@ -0,0 +1,161 @@
+        alias,
+        registerOnly,
+        skipInitialization,
+        publicDeploy,
+        wait,
+        FeeOpts.fromCli(options, log),
+        debugLogger,
+        log,
+      );
+    });
+
+  const deployCommand = program
+    .command('deploy')
+    .description('Deploys a compiled Aztec.nr contract to Aztec.')
+    .argument(
+      '<artifact>',
+      "A compiled Aztec.nr contract's artifact in JSON format or name of a contract artifact exported by @aztec/noir-contracts.js",
+    )
+    .option('--init <string>', 'The contract initializer function to call', 'constructor')
+    .option('--no-init', 'Leave the contract uninitialized')
+    .option('--args <constructorArgs...>', 'Contract constructor arguments', [])
+    .option(
+      '-k, --public-key <string>',
+      'Optional encryption public key for this address. Set this value only if this contract is expected to receive private notes, which will be encrypted using this public key.',
+      parsePublicKey,
+    )
+    .option(
+      '-s, --salt <hex string>',
+      'Optional deployment salt as a hex string for generating the deployment address.',
+      parseFieldFromHexString,
+    )
+    .option('--universal', 'Do not mix the sender address into the deployment.')
+    .option(
+      '-a, --alias-or-address <string>',
+      'Alias or address of the account to deploy from. Incompatible with --private-key.',
+    )
+    .addOption(createPrivateKeyOption("The sender's private key. Incompatible with --alias-or-address", false))
+    .option('--json', 'Emit output as json')
+    // `options.wait` is default true. Passing `--no-wait` will set it to false.
+    // https://github.com/tj/commander.js#other-option-types-negatable-boolean-and-booleanvalue
+    .option('--no-wait', 'Skip waiting for the contract to be deployed. Print the hash of deployment transaction')
+    .option('--no-class-registration', "Don't register this contract class")
+    .option('--no-public-deployment', "Don't emit this contract's public bytecode");
+
+  addOptions(deployCommand, FeeOpts.getOptions()).action(async (artifactPath, _options, command) => {
+    const { deploy } = await import('../cmds/deploy.js');
+    const options = command.optsWithGlobals();
+    const {
+      json,
+      publicKey,
+      args: rawArgs,
+      salt,
+      wait,
+      privateKey,
+      classRegistration,
+      init,
+      publicDeployment,
+      universal,
+      rpcUrl,
+      aliasOrAddress,
+    } = options;
+    await deploy(
+      artifactPath,
+      json,
+      rpcUrl,
+      publicKey ? PublicKeys.fromString(publicKey) : undefined,
+      rawArgs,
+      salt,
+      privateKey,
+      aliasOrAddress,
+      typeof init === 'string' ? init : undefined,
+      !publicDeployment,
+      !classRegistration,
+      typeof init === 'string' ? false : init,
+      universal,
+      wait,
+      FeeOpts.fromCli(options, log),
+      debugLogger,
+      log,
+      logJson(log),
+    );
+  });
```

Comment:
![image](https://github.com/user-attachments/assets/da0f1562-16bf-4d2c-9944-7fc1f08425d6)

On a more serious note, I'd consider a helper fn that takes the options and populates `deploy`'s params (by e.g. inserting defaults to missing values), or at least having the options have the same order as the params.

Reply:
Every command requires some level of cleanup, I agree (including removing a bunch of unnecessary options). They were directly taken from the old CLI bc I wanted to get the main plumbing/refactor in asap so we can expand further. Will tackle this in the next PRs 👍